AsyncRacing/AsyncRacing
# tooling: use prettier for styling instead of eslint

- remove dependency on eslint - we do not need it because react-scripts already depends on it
- remove eslint-airbnb config and all the manual eslint configuration
- add eslint-prettier which disables all eslint rules that can be handled by prettier instead.
- reformat the most important files with the new configuration

What this does is that:
 - Eslint is not used for styling anymore. Only to verify the code is correct (for example that we respect the rules of hooks)
 - Prettier is used for all the styling. You should configure your editor to run prettier every time you save a file (install the vscode-prettier extension)
 

How to test:
- checkout the branch
- run `npm install`
- restart vscode
- install the vscode-prettier extension
- go mess up the formatting in one `.ts[x]` file (for example, change a string quote style or remove spaces at the beginning of the line)
- save the file
- prettier should fix the formatting for you automatically
- try to use a hook outside of a function, eslint should still show an error in your editor
